swap-protocol-home.js

```diff
@@ -0,0 +1,295 @@
+    const cssIdTVL = cssIds[pool].totalValueLocked;
+    setDisplayValueById(cssIdTVL, totalValueLockedUsd);
+
+    totalAssetValue += totalValueLockedByPool[pool].totalValueLocked;
+  }
+
+  const totalAssetValueUsd = usdFormatter.format(totalAssetValue);
+  setDisplayValueById(cssIdTAV, totalAssetValueUsd);
+};
+
+
+const setSwpPrice = async (swpMarketData) => {
+  let swpPriceInUSD = 0;
+  if (swpMarketData) {
+    swpPriceInUSD = swpMarketData.current_price.usd;
+  }
+
+  return {
+    price: swpPriceInUSD
+  };
+};
+
+const setRewardApyDisplayValue = async (pools, siteData, cssIds) => {
+  const prices = siteData['prices'];
+  const swpPoolData = siteData['swpPoolData'];
+  const swpRewardsPerYearByPool = siteData['swpRewardsPerYearByPool'];
+
+  for (const pool of pools) {
+    let tvlAmount = 0;
+    if (swpPoolData[pool].totalValueLocked) {
+      tvlAmount = Number(swpPoolData[pool].totalValueLocked);
+    }
+    const reward = swpRewardsPerYearByPool[pool];
+
+    const totalSharesPerPool = swpPoolData[pool].totalShares;
+    const rewardsPerYearUsd = Number(swpRewardsPerYearByPool[pool]['swp-a'].amount) * prices[commonDenomMapper(reward['swp-a'].denom)].price;
+
+    let rewardsPerShareUsd = 0;
+    let totalLiquidityPerShareUsd = 0;
+    if (totalSharesPerPool) {
+      rewardsPerShareUsd = rewardsPerYearUsd / totalSharesPerPool;
+      totalLiquidityPerShareUsd = tvlAmount / totalSharesPerPool;
+    }
+
+    let rewardApy = 0;
+    if (totalLiquidityPerShareUsd) {
+      const rawApy = (rewardsPerShareUsd / totalLiquidityPerShareUsd) * 100;
+      const apyWithDollarSign = usdFormatter.format(rawApy);
+      rewardApy = formatPercentage(noDollarSign(apyWithDollarSign));
+    }
+
+    const cssId = cssIds[pool].rewardApy;
+    setDisplayValueById(cssId, rewardApy);
+  }
+};
+
+const updateDisplayValues = async(denoms, pools) => {
+  const [
+    pricefeedResponse,
+    incentiveParametersResponse,
+    swpMarketResponse,
+    swpPoolsResponse,
+  ] = await Promise.all([
+    fetch(`${BASE_URL}/pricefeed/prices`),
+    fetch(`${BASE_URL}/incentive/parameters`),
+    fetch('https://api.coingecko.com/api/v3/coins/kava-swap'),
+    fetch(`${BASE_URL}/swap/pools`),
+  ]);
+
+  const swpMarketJson = await swpMarketResponse.json();
+  const swpPoolDataJson = await swpPoolsResponse.json();
+
+  let siteData = {};
+  const cssIds = mapCssIds(pools);
+
+  const pricefeedPrices = await pricefeedResponse.json();
+  const incentiveParamsJson = await incentiveParametersResponse.json();
+
+  const prices = await mapPrices(denoms, pricefeedPrices.result);
+  siteData['prices'] = prices;
+
+  const denomConversions = setConversionFactors(denoms);
+  siteData['denomConversions'] = denomConversions;
+
+  const incentiveParams = await incentiveParamsJson.result;
+  siteData['incentiveParams'] = incentiveParams;
+
+  const swpPrice = await setSwpPrice(swpMarketJson.market_data);
+  siteData['prices']['swp-a'] = swpPrice;
```

Comment:
Swp price comes from different API than the other pricefeeds, so it needs to be added to `siteData`.